### PR TITLE
add tests for downloading empty file/dir to fs

### PR DESCRIPTION
**Pull request by joehand**
Adds (currently failing) tests for downloading empty file/directory to file system. #28

### CODE REVIEW

**Comment by karissa** [comment]
+1 nice

File: test/replicates.js [hunk 1]
```diff
@@ -111,3 +113,86 @@ tape('replicates empty directories', function (t) {
     stream.pipe(streamClone).pipe(stream)
   })
 })
+
+tape('downloads empty files to fs', function (t) {
+  var drive = hyperdrive(memdb())
+  var driveClone = hyperdrive(memdb())
+  var tmp = path.join(os.tmpdir(), 'hyperdrive-test')
+  mkdirp.sync(tmp)
```
Comment (juliangruber) [inline test/replicates.js]: we don't need mkdirp for that, os.tmpdir should already exist

File: test/replicates.js [hunk 2]
```diff
@@ -111,3 +113,86 @@ tape('replicates empty directories', function (t) {
...
+
+  var archive = drive.createArchive()
+
+  var ws = archive.createFileWriteStream('empty.txt')
+  ws.end()
```
Comment (juliangruber) [inline test/replicates.js]: can be refactored to `archive.createFileWriteStream('empty.txt').end()`. i was looking where else `ws` was used because it got a name

**Comment by juliangruber** [comment]
after the comments, +1!

**Comment by joehand** [comment]
Thanks @juliangruber, updated with your comments.

**Comment by juliangruber** [comment]
+1!

**Comment by clkao** [comment]
Do we want to make this as todo test and merge?  It seems we need to either have raf or additional storage api to support mkdir.

**Comment by clkao** [comment]
well marking as todo requires https://github.com/substack/tape/pull/254

**Comment by mafintosh** [comment]
@clkao use tape.skip and add a todo comment for now. mkdir will be a hyperdrive option where you can pass a mkdir handler in hyperdrive

**Comment by joehand** [comment]
@mafintosh changed the tests to skip.
